Dockerfile.buf

```
@@ -7,14 +7,18 @@ RUN go mod download
 
 COPY cmd /workspace/cmd
 COPY internal /workspace/internal
-RUN CGO_ENABLED=0 go build -ldflags "-s -w" -trimpath -o /go/bin/buf ./cmd/buf
 
-FROM alpine:3.12.4
+ARG TARGETOS
+ARG TARGETARCH
+RUN CGO_ENABLED=0 GOOS=${TARGETOS} GOARCH=${TARGETARCH} \
+  go build -ldflags "-s -w" -trimpath -o /go/bin/buf ./cmd/buf
+
+FROM --platform=${BUILDPLATFORM} alpine:3.12.4
```

If I'm understanding how the buildx command works correctly, this final stage should be from the target platform

```suggestion
FROM --platform=${TARGETPLATFORM} alpine:3.12.4
```

---

According to [Docker Buildx | Docker Documentation](https://docs.docker.com/buildx/working-with-buildx/#build-multi-platform-images)

> Finally, depending on your project, the language that you use may have good support for cross-compilation. In that case, multi-stage builds in Dockerfiles can be effectively used to build binaries for the platform specified with --platform using the native architecture of the build node. A list of build arguments like BUILDPLATFORM and TARGETPLATFORM is available automatically inside your Dockerfile and can be leveraged by the processes running as part of your build.

```
FROM --platform=$BUILDPLATFORM golang:alpine AS build
ARG TARGETPLATFORM
ARG BUILDPLATFORM
RUN echo "I am running on $BUILDPLATFORM, building for $TARGETPLATFORM" > /log
FROM alpine
COPY --from=build /log /log
```

---

Yes, you are right.

Reference:
1. https://docs.docker.com/engine/reference/builder/#from
2. https://docs.docker.com/buildx/working-with-buildx/#build-multi-platform-images